iox_query/src/frontend/influxql.rs

```diff
@@ -43,6 +51,61 @@ impl SchemaProvider for ContextSchemaProvider {
     }
 }
 
+/// A physical operator that overrides the `schema` API,
+/// to return an amended version owned by `SchemaExec`. The
+/// principal use case is to add additional metadata to the schema.
+struct SchemaExec {
+    input: Arc<dyn ExecutionPlan>,
+    schema: SchemaRef,
+}
+
+impl Debug for SchemaExec {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "SchemaExec")
+    }
+}
+
+impl ExecutionPlan for SchemaExec {
+    fn as_any(&self) -> &dyn Any {
+        self
+    }
+
+    fn schema(&self) -> SchemaRef {
+        Arc::clone(&self.schema)
+    }
+
+    fn output_partitioning(&self) -> Partitioning {
+        self.input.output_partitioning()
+    }
+
+    fn output_ordering(&self) -> Option<&[PhysicalSortExpr]> {
+        self.input.output_ordering()
+    }
+
+    fn children(&self) -> Vec<Arc<dyn ExecutionPlan>> {
+        unimplemented!()
+    }
+
+    fn with_new_children(
+        self: Arc<Self>,
+        _children: Vec<Arc<dyn ExecutionPlan>>,
+    ) -> Result<Arc<dyn ExecutionPlan>> {
+        unimplemented!()
```

Comment:
Likewise here I think it would be good to avoid the panic 
Something like (untested):

```suggestion
assert!(children.len(), 1);
Ok(Arc::new(Self {
  schema: Arc::clone(&self.schema),
  input: children.pop(),
}))
```